File: GeneratedEntityReaderClasses/Components/MaterialAreaCheckerComponent.cs
namespace GeneratedNoitaClasses;
public class MaterialAreaCheckerComponent : NoitaComponentBase,  iNoitaType<MaterialAreaCheckerComponent>{
	public NoitaBool Deleted { get; set; } = new();
	public NoitaBool Enabled { get; set; } = new();
	public NoitaString Tags { get; set; } = new();
	public NoitaInt update_every_x_frame { get; set; } = new();
	public NoitaBool look_for_failure { get; set; } = new();
	public types_aabb area_aabb { get; set; } = new();
	public NoitaInt material { get; set; } = new();
	public NoitaInt material2 { get; set; } = new();
	public NoitaInt count_min { get; set; } = new();
	public NoitaBool always_check_fullness { get; set; } = new();
	public NoitaBool kill_after_message { get; set; } = new();

    public void Read(NoitaStream s){
		Deleted.Read(s);
		Enabled.Read(s);
		Tags.Read(s);
		update_every_x_frame.Read(s);
		look_for_failure.Read(s);
		area_aabb.Read(s);
		material.Read(s);
		material2.Read(s);
		count_min.Read(s);
		always_check_fullness.Read(s);
		kill_after_message.Read(s);

    }
    public void Write(NoitaStream s){
		Deleted.Write(s);
		Enabled.Write(s);
		Tags.Write(s);
		update_every_x_frame.Write(s);
		look_for_failure.Write(s);
		area_aabb.Write(s);
		material.Write(s);
		material2.Write(s);
		count_min.Write(s);
		always_check_fullness.Write(s);
		kill_after_message.Write(s);

    }
}

File: GeneratedEntityReaderClassesPython/GeneratedClasses/Components/MaterialAreaCheckerComponent.py
from dataclasses import dataclass, field
from BasicNoitaTypes import NoitaBool
from BasicNoitaTypes import NoitaString
from BasicNoitaTypes import NoitaInt
from GeneratedClasses.types_aabb import types_aabb

from NoitaStream import NoitaStream
@dataclass
class MaterialAreaCheckerComponent:
    Deleted:NoitaBool = field(default_factory=NoitaBool)
    Enabled:NoitaBool = field(default_factory=NoitaBool)
    Tags:NoitaString = field(default_factory=NoitaString)
    update_every_x_frame:NoitaInt = field(default_factory=NoitaInt)
    look_for_failure:NoitaBool = field(default_factory=NoitaBool)
    area_aabb:types_aabb = field(default_factory=types_aabb)
    material:NoitaInt = field(default_factory=NoitaInt)
    material2:NoitaInt = field(default_factory=NoitaInt)
    count_min:NoitaInt = field(default_factory=NoitaInt)
    always_check_fullness:NoitaBool = field(default_factory=NoitaBool)
    kill_after_message:NoitaBool = field(default_factory=NoitaBool)
    
    def Read(self, s:NoitaStream):
        self.Deleted.Read(s)
        self.Enabled.Read(s)
        self.Tags.Read(s)
        self.update_every_x_frame.Read(s)
        self.look_for_failure.Read(s)
        self.area_aabb.Read(s)
        self.material.Read(s)
        self.material2.Read(s)
        self.count_min.Read(s)
        self.always_check_fullness.Read(s)
        self.kill_after_message.Read(s)
    
    def Write(self, s:NoitaStream):
        self.Deleted.Write(s)
        self.Enabled.Write(s)
        self.Tags.Write(s)
        self.update_every_x_frame.Write(s)
        self.look_for_failure.Write(s)
        self.area_aabb.Write(s)
        self.material.Write(s)
        self.material2.Write(s)
        self.count_min.Write(s)
        self.always_check_fullness.Write(s)
        self.kill_after_message.Write(s)

